### PR TITLE
istioctl analyze ConflictingGateways no match subselector

### PR DESCRIPTION
**Pull request by nicole-lihui**
**Please provide a description of this PR:**

when gateways selectors no match sub selectors(point to the same gateway instances)

### CODE REVIEW

File: pkg/config/analysis/analyzers/testdata/conflicting-gateways-subSelector.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    hosts:
+    - "foo.bar"
+---
+apiVersion: networking.istio.io/v1
+kind: Gateway
+metadata:
+  name: alpha
+spec:
+  selector:
+    istio: ingressgateway
```
Comment (MorrisLaw) [inline pkg/config/analysis/analyzers/testdata/conflicting-gateways-subSelector.yaml:24]: Does it not matter if `app: istio-ingressgateway` isn't included here in the selector?

Comment (nicole-lihui) [reply pkg/config/analysis/analyzers/testdata/conflicting-gateways-subSelector.yaml:24]: Hit the nail on the head

Comment (nicole-lihui) [reply pkg/config/analysis/analyzers/testdata/conflicting-gateways-subSelector.yaml:24]: we shoud judge based on the instances hit

**Comment by nicole-lihui** [comment]
@MorrisLaw PTAL

**Review by MorrisLaw — APPROVED** [review]

**Pull request merged**